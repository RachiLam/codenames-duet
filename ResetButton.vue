<template>
  <button @click="reset" />
</template>

<script>
export default {
  methods: {
    reset() {
      window.location.reload();
    },
  },
};
</script>

<style lang="scss" scoped>
  @import './variables';

  button {
    width: $reset-button-size;
    height: $reset-button-size;
    background-color: $accent-light;
    border-radius: 25%;
  }
</style>
