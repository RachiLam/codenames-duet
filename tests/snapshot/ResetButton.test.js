import { shallowMount } from '@vue/test-utils';
import ResetButton from '../../src/components/ResetButton.vue';

test('renders as inactive', () => {
  const wrapper = shallowMount(ResetButton, {
    propsData: {
      activated: false,
    },
  });
  expect(wrapper.element).toMatchSnapshot();
});

test('renders as activated', () => {
  const wrapper = shallowMount(ResetButton, {
    propsData: {
      activated: true,
    },
  });
  expect(wrapper.element).toMatchSnapshot();
});
