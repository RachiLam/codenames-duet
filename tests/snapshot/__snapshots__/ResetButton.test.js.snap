// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as activated 1`] = `
<button
  class="activated"
/>
`;

exports[`renders as inactive 1`] = `
<button
  class=""
/>
`;
